### DISCUSSION TITLE
Sqlserver connection string problem

### DISCUSSION THREAD

**Discussion by JakovGlavac** [prisma/prisma#6936] [Q&A] [answered]
Hi,
I created a database instance on Azure, and now I'm trying to connect to it. 
I tried to use  `sqlserver://HOST:PORT;database=DATABASE;user=USER;password=PASSWORD` format but it didn't work.

Then I found that azure can autogenerate JDBC connection string like this:
![image](https://user-images.githubusercontent.com/34971091/117084388-6a249e00-ad47-11eb-9711-856073324e70.png)

I found this comment in which @pimeys says that the connection string for Prisma is the same as JDBC minus `jdbc:` https://github.com/prisma/prisma/issues/6845#issuecomment-829111480.

So I did that and got an error  `Error validating: This line is not a valid definition within a datasource.`

![image](https://user-images.githubusercontent.com/34971091/117084345-3ea1b380-ad47-11eb-8303-b6c04b8735d6.png)

What I'm missing?

**Comment by JakovGlavac** [comment, marked as answer]
I solved it!
The problem was that my password was `LGC]r]>t!M34\8up`, and for some reason, Prisma couldn't parse it, and that's why I was getting that error.
I think this is a bug because the SQL server allows having those characters as a password.

**Reply by pimeys** [reply]
You can escape in JDBC with `{}`. So you could do `password={LGC]r]>t!M34\8up}` and it should work.

**Reply by pimeys** [reply]
But, I see you have it in `{}`... I'll take a look.

**Reply by pimeys** [reply]
I think I found your problem with the previous password. It has a `\` character, which is then in your `.env` and your terminal that reads it cannot parse it correctly. The proper way to write that old password would've been `{LGC]r]>t!M34\\8up}` with `\\` so the terminal then removes the other one and sends us the correct string with only one `\`.

**Reply by JakovGlavac** [reply]
Yeah but in schema.prisma its the same problem

**Reply by luomavaltteri** [reply]
Hi. I was fighting this same issue for a day. We have Azure SQL database and couldn't figure out how to escape the connection string properly. 

Thank you @pimeys for mentioning out that you can escape the usernames and passwords with {} in JDBC. 
Could this be a thing that might be relevant to mention in Prisma's SQL Server guide?

**Reply by janpio** [reply]
cc @mhwelander

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
